src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs

```diff
@@ -70,4 +59,20 @@ internal static void DisposeWithException(this Activity activity, Exception exce
             }
         }
     }
+
+    private static Activity StartActivityWithTags(this ActivitySource activitySource, string operationName, ActivityKind kind, ITags tags)
+    {
+        var activity = activitySource.StartActivity(operationName, kind);
+
```

Comment:
How about adding

```suggestion
        var activity = activitySource.StartActivity(operationName, kind);
        if (activity == null)
        {
            return null;
        }
```

to reduce the possibility of throwing a NullPointerException? 

It should fix https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1282

I think we should be very careful and make sure we never throw an exception at runtime.

Reply:
Sure, lets put it into separate PR. I know that it will be small, but maybe there will be a chance to somehow test it.